src/faults/AbstractPerturbation.jl

```diff
@@ -48,21 +48,26 @@ function simulate(np::AbstractPerturbation, powergrid::PowerGrid, x1::Array, tim
     @assert np.tspan_fault[2] <= last(timespan) "fault cannot end in the future"
 
     np_powergrid = np(powergrid)
+    regular = rhs(powergrid)
+    error = rhs(np_powergrid)
+    _f = (dx, x, p, t) -> p ? regular(dx,x,p,t) : error(dx,x,p,t)
+
+    f = ODEFunction(_f, mass_matrix = regular.mass_matrix, syms = regular.syms)
```

Comment:
Are there abstract pertubations which change the size of the system/the mass matrix?

Reply:
In theory yes. `LineFailure` is a subtype of `AbstactPerturbation`. So if we have a system with dynamic lines, this failure would indeed change the system size. There is no test for this specific situation so far and I am not sure this even worked before.
However, I am not sure how meaningful it would be to actually support that case. If you go down to model the electromagnetic transients in your line, you would maybe also need a more detailed model for the failure than just a vanishing line from one moment to the other.

Reply:
I've asked on slack an there is no way to change the mass matrix yet. So it is not really possible to `resize!` DAE integrators right now. 
This would also complicate things a lot since the mapping of symbols to indices in the state vector would go bad. IMO topology changes should be modeled within the original topology: i.e. set the current of a vanished line to 0.
Maybe we could add a check that there is no dimension missmatch between error and normal state.